libs/hdf-converters/src/snyk-mapper.ts

```diff
@@ -37,132 +40,580 @@ function nistTag(identifiers: unknown[]): string[] {
   );
 }
 
+type SnykUserInputDatum = {
+  organization: string;
+  project?: string;
+};
+
+type SnykDatum = SnykUserInputDatum & {
+  organizationData: Record<string, unknown>;
+  project: string;
+  projectData: Record<string, unknown>;
+  issues: Record<string, unknown>[];
+  dependencyGraph: Record<string, unknown>;
+  dependencyChains: Record<string, string[][]>;
+};
+
 export class SnykResults {
-  data: Record<string, unknown>;
-  customMapping?: MappedTransform<ExecJSON.Execution, ILookupPath>;
-  constructor(snykJson: string) {
-    this.data = JSON.parse(snykJson);
+  data?: Record<string, unknown>[] | SnykDatum[];
+  withRaw: boolean;
+
+  apiInput?: SnykUserInputDatum[];
+  usingApi = false;
+  apiToken?: string;
+  apiVersion?: 1; // can be 1 or (to be implemented) 3
+  apiClient?: AxiosInstance;
+
+  constructor(snykJson?: string, withRaw = false) {
+    if (snykJson !== undefined) {
+      this.data = JSON.parse(snykJson);
+    }
+    if (this.data !== undefined && !Array.isArray(this.data)) {
+      this.data = [this.data];
+    }
+    this.withRaw = withRaw;
   }
 
-  toHdf(): ExecJSON.Execution[] | ExecJSON.Execution {
-    const results: ExecJSON.Execution[] = [];
-    if (Array.isArray(this.data)) {
-      this.data.forEach((element) => {
-        const entry = new SnykMapper(element);
-        if (this.customMapping !== undefined) {
-          entry.setMappings(this.customMapping);
+  // api constructor
+  static SnykAPI(
+    apiToken: string,
+    apiVersion: 1 = 1,
```

Comment:
```suggestion
    apiVersion: number = 1,
```

Reply:
This is intentional here.  When we add support for api version 3 it'll be `apiVersion: 1|3 = 1`.  I'm more less using it as an enum.